### ISSUE TITLE
[Bug]: When saving a document with inherited areas, the editables in the settings dialog are saved, even if not overridden

### ISSUE THREAD

**Issue by ThisIsJustARandomGuy** [pimcore/pimcore#12775]
### Expected behavior

When using a content master document, I expect the editables in options dialogs to be inherited from that document even after saving the child document.

### Actual behavior

Inheritance works correctly for "normal" editables. If I don't right-click and select "override", their content is correctly taken from the content master document. However, the same is not true for the editables in the options dialogs.

When saving, I think they are not considered as inherited. They are stored as if they were overridden in the child document when in reality they are not. This is a problem since it effectively breaks the ability to have just the content translated and leave the options in the content master document.

However, when saving an inherited document, the values in the options dialog are saved even if they are not overridden.

### Steps to reproduce

1. Create a new page `/master` and add an area which has an options dialog and some content. For example, let's take a `Headline` area. It returns an `align` select from `getAdditionalConfig()` and has a `pimcore_input('text')` in the template). You can do this in the official demo if you want.
2. Set some content and some settings and then save the page. The request payload will look something like the following. In my example, it correctly contains the option `align` as well as the content `text`:
```JSON
{
    "content:1.align": {
        "data":"center",
        "type":"select"
    },
    "content:1.text": {
        "data":"My headline",
        "type":"input"
    }
}
```
3. Create a second page `/child`. Go to the "SEO & Editables" tab
4. Set `/master` as the Content Master Document and click "apply". *Do NOT click `Save` or `Save & Publish`, as this is where the bug occurs.*
5. Let's verify that inheritance works correctly. Go back to `/master` and make some changes to the settings and content of the area added in step 1.
6. Save & Publish `/master`. Again, the `save` request correctly contains the `align` and `text` editables just like in step 2.
7. Now go back to `/child` and reload it. As expected, the changed setting and content were applied correctly.
8. Now click "Save & Publish" on `/child`. The payload will look something like this:
```JSON
{
    "content:1.align":{
        "data":"center",
        "type":"select"
    }
}
```
The payload should *not* contain `content:1.align`, as that editable is inherited from the options dialog in `/master` and we never overrode it in `/child`. Notice that it does not contain `content:1.text` which is correct, since we didn't override that as well.
9. Go back `/master` and make the same changes as in step 5. Once again, the payload correctly contains both editables.
10. However, if you reload the second document now, you'll see that only the content change in `text` is inherited correctly and the options like `align` stay at whatever they were in step 8.

The only way to undo this is to either re-apply the content master document (thus losing all the changes already made) or manually carrying over 

TL;DR: Create a document, apply a content master document to it and click "Save" or "Save & Publish". From that point on, settings in the optoins dialog aren't inherited anymore and changes from the content master document aren't applied.

**Comment by kingjia90** [comment]
Fixed by #12811
